activemodel/lib/active_model/validations/length.rb

```diff
@@ -15,7 +15,10 @@ class LengthValidator < EachValidator # :nodoc:
       def initialize(options)
         if range = (options.delete(:in) || options.delete(:within))
           raise ArgumentError, ":in and :within must be a Range" unless range.is_a?(Range)
-          options[:minimum], options[:maximum] = range.min, range.max
+          options[:minimum] = range.min if range.begin?
+          if range.end?
+            options[:maximum] = (range.exclude_end? ? range.end - 1 : range.end)
+          end
```

Comment:
Do we really need to check `infinite?`?  A range with `range.begin.infinite?` isn't really valid from length check perspective.  Such a length check would always fail.  And if `range.end.infinite?`, the length would never exceed the maximum, so the length check would always pass as intended.  

```suggestion
          options[:minimum] = range.min if range.begin
          options[:maximum] = (range.exclude_end? ? range.end - 1 : range.max) if range.end
```

It's also worth pointing out that `range.max` raises an error for beginless, end-excluded ranges:

```ruby
(..10).max
# => 10

(...10).max
# => cannot exclude end value with non Integer begin value (TypeError)
```

Which necessitates `range.end - 1` instead of `range.max - 1` (though I've kept `range.max` in the `false` branch to make the distinction).

Reply:
Yes, this works, thanks 👍 
I checked only for `..10`-style ranges and got an error for `max/min`, but, turns out, it is working for `-Float::INFINITY..10`-style ranges 🙄 
In other PR regarding postgres `tsrange` type similar problem - for one style works, for other is not. This differences sucks.

So regarding core extension: yay or nay?

Reply:
> but, turns out, it is working for `-Float::INFINITY..10`-style ranges

Let's add some regression tests for different combinations (e.g. `0..Float::INFINITY`, `0...Float::INFINITY`, `..Float::INFINITY`, `-Float::INFINITY..10`, `-Float::INFINITY..`, etc).  We can just loop over them instead of having separate test cases for each.  

> So regarding core extension: yay or nay?

It would need Core Team approval.  My own inclination would be to wait and see how https://bugs.ruby-lang.org/issues/15864 plays out.  Although those extension methods aren't required for this PR, if you can demonstrate other cases where they are helpful, that might push the Ruby issue forward.